scipy/optimize/_differentialevolution.py

```diff
@@ -742,11 +757,20 @@ def maplike_for_vectorized_func(func, x):
         else:
             self.integrality = False
 
+        # check for equal bounds
+        eb = self.limits[0] == self.limits[1]
+        eb_count = np.count_nonzero(eb)
+
         # default population initialization is a latin hypercube design, but
         # there are other population initializations possible.
         # the minimum is 5 because 'best2bin' requires a population that's at
         # least 5 long
-        self.num_population_members = max(5, popsize * self.parameter_count)
+        # 202301 - reduced population size to account for parameters with
+        # equal bounds. If there are no varying parameters set N to at least 1
```

Comment:
Just to make sure I understand correctly - if there are no free parameters, this will still do some work but terminate normally with the fixed values as the solution. The alternative would be to simply return after detecting that all parameters are fixed, but perhaps the way you have it now avoids the hassle of setting the other fields of the `OptimizeResult` appropriately (e.g. `success` depending on whether constraints are satisfied)?

Reply:
If there are no free parameters then the minimal amount of work will be done. i.e. the population will be the smallest it can be, and the optimisation should stop after one iteration. I took this route so that the output would be as if the optimiser had continued to run. For example, one should still go through the polishing process, figure out what the constraint violations are, etc.
Implementing a shortcut would involve figuring out all that logic, and make the internals messier. For an edge case that's unlikely to happen I didn't think it was worth the extra work.

Reply:
I agree. Sounds good. Just wanted to make sure it would not do too much work.